### PR TITLE
Unifying the endpoints for contact flows

### PR DESCRIPTION
**Pull request by MishkaRogachev**
For https://github.com/status-im/status-desktop/issues/10158
Depends: https://github.com/status-im/status-go/pull/3327

Remove legacy contact request (was used for desktop 0.8) 

Important changes:
1. From https://github.com/status-im/status-go/pull/3327:
 - [x] Outgoing CR should invoke notification with pending state on the sender's side
 - [x] Fix `From` for outgoing PENDING contact requests
 - [x] Fix `From` for outgoing ACCEPTED contact requests
 - [x] Fix displaying CR in regular chat
 - [x] No username and other contact details for outgoing pending CR notification (happens sometimes)
 2. Fresh fixes:
 - [x] Chat appears for both users after becoming a mutual contacts
 - [x] No default CR message in chat should appear
 - [x] Outgoing notification is not being updated using `addContact`. (switched to `sendContactReequest`)
 - [x] Test refactored to use common steps for contact request flows

### CODE REVIEW

**Review by ilmotta — APPROVED** [review]
@MishkaRogachev, I tested some CR flows in mobile pointing to this branch, and I didn't see regressions. As we discussed in private, mobile should be changed to stop using the `wakuext_addContact` endpoint, but meanwhile, this PR LGTM.

File: protocol/messenger_handler.go [hunk 1]
```diff
@@ -265,8 +265,33 @@ func (m *Messenger) PendingNotificationContactRequest(contactID string) (*Activi
...
+		messageState.CurrentMessageState.WhisperTimestamp,
+		contact,
+		"Please add me to your contacts",
+		false,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	contactRequest.ID = "0x" + types.Bytes2Hex(append(types.Hex2Bytes(contact.ID), 0x20))
```
Comment (cammellos) [inline protocol/messenger_handler.go]: Is this `defaultContactRequestID`?

Comment (MishkaRogachev) [reply protocol/messenger_handler.go]: Previously there were two meanings of default contact request, this one left for one created by handling ContactUpdate. Should i return `defaultContactRequestID ` function?

**Review by cammellos — APPROVED** [review]

File: protocol/messenger_handler.go (hunk 1 above)
Comment (MishkaRogachev) [reply protocol/messenger_handler.go]: fixed

**Review by alwx — APPROVED** [review]

File: protocol/messenger_contacts.go [hunk 2]
```diff
@@ -24,7 +24,29 @@ func (m *Messenger) acceptContactRequest(ctx context.Context, requestID string,
...
+
+		chat = OneToOneFromPublicKey(publicKey, m.getTimesource())
+	}
+
+	chat.Active = true
+	if err := m.saveChat(chat); err != nil {
+		return nil, err
+	}
+
+	return response, nil
```
Comment (cammellos) [inline protocol/messenger_contacts.go:50]: you are not adding the chat to the response

Comment (MishkaRogachev) [reply protocol/messenger_contacts.go:50]: fixed

**Pull request merged**